x-pack/plugins/security/server/authorization/index.ts

```diff
@@ -112,8 +112,8 @@ export function setupAuthorization({
         authz
       );
 
-      // if we're an anonymous route, we disable all ui capabilities
-      if (request.route.options.authRequired === false) {
+      // if we're an anonymous route or optional auth, we disable all ui capabilities
+      if (!http.auth.isAuthenticated(request)) {
```

Comment:
@legrego I updated this code and https://github.com/elastic/kibana/blob/d66b3c74eb18d72208d0652a0b5f72cd9701191b/x-pack/test/ui_capabilities/spaces_only/tests/catalogue.ts#L33 started failing because security is disabled there. should I update the test or revert these changes?

Reply:
@restrry I'm not seeing the test fail locally. I would expect all capabilities to be disabled in this scenario (so the test should be correct as-is)

Reply:
well...it's green now...oookay. sorry for noize